File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/receive/mvp/ReceivePresenter.java
package com.github.willjgriff.ethereumwallet.ui.receive.mvp;

import com.github.willjgriff.ethereumwallet.ethereum.account.AddressManager;
import com.github.willjgriff.ethereumwallet.mvp.BaseMvpPresenter;

import javax.inject.Inject;

/**
 * Created by Will on 01/03/2017.
 */

public class ReceivePresenter extends BaseMvpPresenter<ReceiveView> {

	private AddressManager mAddressManager;

	@Inject
	public ReceivePresenter(AddressManager addressManager) {
		mAddressManager = addressManager;
	}

	@Override
	public void viewReady() {
		getView().setReceiveAddress(mAddressManager.getActiveAddress().getHex());
	}
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/receive/mvp/ReceivePresenter.kt
package com.github.willjgriff.ethereumwallet.ui.receive.mvp

import com.github.willjgriff.ethereumwallet.ethereum.address.AddressManager
import com.github.willjgriff.ethereumwallet.ethereum.address.balance.AddressBalance
import com.github.willjgriff.ethereumwallet.mvp.BaseMvpPresenterKotlin
import javax.inject.Inject

/**
 * Created by williamgriffiths on 18/04/2017.
 */
class ReceivePresenter @Inject constructor(private val addressManager: AddressManager, private val addressBalance: AddressBalance)
    : BaseMvpPresenterKotlin<ReceiveView>() {

    override fun viewReady() {
        view?.setReceiveAddress(addressManager.getActiveAddress().hex)

        addDisposable(addressBalance.getPendingBalanceAtActiveAddress()
                .subscribe { view?.setPendingBalance(it) })

        addDisposable(addressBalance.getBalanceAtActiveAddress()
                .subscribe { view?.setConfirmedBalance(it) })
    }
}